File: leetcode/stacks/valid_parentheses/python/test_solution.py
import unittest

from solution import solve


class ValidParenthesesTests(unittest.TestCase):
    def test_examples(self) -> None:
        self.assertTrue(solve("()"))
        self.assertTrue(solve("()[]{}"))
        self.assertFalse(solve("(]"))
        self.assertFalse(solve("([)]"))
        self.assertFalse(solve("("))


if __name__ == "__main__":
    unittest.main()

File: leetcode/stacks/valid_parentheses/js/test_solution.js
const assert = require('assert');
const { solve } = require('./solution');

assert.strictEqual(solve("()"), true);
assert.strictEqual(solve("()[]{}"), true);
assert.strictEqual(solve("(]"), false);
assert.strictEqual(solve("([)]"), false);
assert.strictEqual(solve("("), false);
console.log('OK');
